telemetry_package/main.py

```diff
@@ -1,31 +1,140 @@
-import sys
+'''
+This code gets too many messages all at once. Hence it needs to: 
+    - The Node only sends the first message it receives
+'''
+
+
+import json
+import os
+import select
+import socket
+# import sys
 import rclpy
 from rclpy.node import Node
+from rcl_interfaces.msg import Log
+
+
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
+
+
+class MinimalPublisher(Node):
 
-from common_package_py.common_node import CommonNode
+    def __init__(self):
+        super().__init__('telemetry_node')
+        self.subscription = self.create_subscription(
+            Log,
+            '/rosout',
+            self.run_server, 
+            100)
+        self.subscription 
 
+        # Counter for debugging purposes
+        self.msg_cnt = 0
 
-class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
+        self.msg_cache = list()
```

Comment:
Why do you implement a cache? I don't think this is necessary.

Reply:
I implemented a cache as a countermeasure to "Errno[32] Broken Pipe" that I was getting while implementing the code. Upon rearranging how the socket gets set up, the cache is in fact *not* necessary.